jbpm-human-task/jbpm-human-task-workitems/src/main/java/org/jbpm/services/task/wih/AbstractHTWorkItemHandler.java

```diff
@@ -224,12 +224,17 @@ protected ContentData createTaskContentBasedOnWorkItemParams(KieSession session,
         return data;
     }
     
-    protected boolean isAutoClaim(WorkItem workItem, Task task) {
-        String swimlaneUser = (String) workItem.getParameter("SwimlaneActorId");
-        if (swimlaneUser != null  && !"".equals(swimlaneUser) && task.getTaskData().getStatus() == Status.Ready) {
-            return true;
-        }
+    protected boolean isAutoClaim(KieSession session, WorkItem workItem, Task task) {
+        String autoclaim = (String) session.getEnvironment().get("Autoclaim");
         
+        if(autoclaim != null && !"".equals(autoclaim) && autoclaim.equals("false")) {
```

Comment:
I think !"".equals(autoclaim) is not needed here since it is trivially true because autoclaim.equals("false") has to be true.

Reply:
Done